precompiles/erc20/integration_test.go

```diff
@@ -179,7 +181,405 @@ var _ = Describe("ERC20 Extension -", func() {
 		_ = passCheck
 	})
 
-	Context("basic functionality -", func() {})
+	Context("basic functionality -", func() {
+		When("transferring tokens", func() {
+			DescribeTable("it should transfer tokens to a non-existing address", func(callType CallType, expGasUsed int64) {
+				sender := is.keyring.GetKey(0)
+				receiver := utiltx.GenerateAddress()
+				fundCoins := sdk.Coins{sdk.NewInt64Coin(is.tokenDenom, 300)}
+				transferCoins := sdk.Coins{sdk.NewInt64Coin(is.tokenDenom, 100)}
+
+				// Fund account with some tokens
+				is.fundWithTokens(callType, contractsData, sender.Addr, fundCoins)
+
+				// Transfer tokens
+				txArgs, transferArgs := is.getTxAndCallArgs(callType, contractsData, erc20.TransferMethod, receiver, transferCoins[0].Amount.BigInt())
+
+				transferCheck := passCheck.WithExpEvents(erc20.EventTypeTransfer)
+
+				res, _, err := is.factory.CallContractAndCheckLogs(sender.Priv, txArgs, transferArgs, transferCheck)
+				Expect(err).ToNot(HaveOccurred(), "unexpected result calling contract")
+
+				is.ExpectBalancesForContract(
+					callType, contractsData,
+					[]ExpectedBalance{
+						{address: sender.AccAddr, expCoins: fundCoins.Sub(transferCoins...)},
+						{address: receiver.Bytes(), expCoins: transferCoins},
+					},
+				)
```

Comment:
curious if you ran into some differences in expected balance due to some of it being used for gas?

Reply:
Ahh on second thought you are not using the native denom so the calculations work!